### PR TITLE
draft: move SPLIT_REDUCEOP into kernel.py

### PR DESCRIPTION
**Pull request by cpwrs**
draft PR for bounty #5542

```
kernel.apply_opt(OptOps.SPLIT, axis, amt)
k1, k2 = kernel.split()
```
Now working on kernels splitting during search.

To do
- [X] Write a test with a hand coded LazyOp with a KERNEL in the middle. Get the lowerer generating the right thing.
- [X] Add support for 0-n ExecItems for each ScheduleItem.
- [X] lazy.py reduce split -> OptOps.SPLIT
- [ ] Add support for varying kernel count in search.

### CODE REVIEW

**Comment by geohot** [comment]
Good luck!

**Comment by cpwrs** [comment]
Thanks for locking!
Should I support n KERNELs in a LazyOp?

**Comment by geohot** [comment]
Yes. You also need to support the split as an OptOps, should be easy once it's in Kernel.

**Comment by geohot** [comment]
Still active?

**Comment by cpwrs** [comment]
Pushed progress, still no good splits. The store shape is wrong in the first reduce. I have time to work on this today.

`fixup_ast` is now adding a correct second kernel to the AST.

**Comment by geohot** [comment]
Still active?

File: tinygrad/engine/realize.py [hunk 1]
```diff
@@ -186,26 +189,30 @@ def run(self, var_vals:Optional[Dict[Variable, int]]=None, wait=False, jit=False
       self.prg.first_run = False
     return et
 
-def lower_schedule_item(si:ScheduleItem) -> ExecItem:
+def lower_schedule_item(si:ScheduleItem) -> Tuple[ExecItem, ...]:
```
Comment (geohot) [inline tinygrad/engine/realize.py:192]: Why Tuple and not List?

**Comment by cpwrs** [comment]
Closing, don't have time to be active on it this week.

**Pull request closed without merging**